[Scheduler] add `JitterTrigger`

kbond opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | #49806
| License       | MIT
| Doc PR        | n/a

I think this is what @vtsykun was trying to accomplish with #49806. FreeBSD has a `-jitter` option that does the same thing so I think there's precedent. From the (https://man.freebsd.org/cgi/man.cgi?cron):

> -j	jitter
>	     Enable time jitter.  Prior	to executing commands, cron will sleep
	     a random number of	seconds	in the range from 0 to jitter.	~This
	     will not affect superuser jobs (see -J).~  A value for jitter must
	     be	between	0 and 60 inclusive.  Default is	0, which effectively
	     disables time jitter.
>	     This option can help to smooth down system	load spikes during mo-
	     ments when	a lot of jobs are likely to start at once, e.g., at
	     the beginning of the first	minute of each hour.

Usage:

```php
// defaults to 0-60 seconds
$schedule->add(RecurringMessage::cron('@daily', $message1)->withJitter());

// customize the max seconds (0-30)
$schedule->add(RecurringMessage::cron('@daily', $message1)->withJitter(30));
```

kbond committed:
[Scheduler] add `JitterTrigger`

Co-authored-by: Uladzimir Tsykun <tsykun314@gmail.com>

kbond commented on src/Symfony/Component/Scheduler/Trigger/CronExpressionTrigger.php:
```diff
@@ -73,9 +76,26 @@ public static function fromSpec(string $expression = '* * * * *', string $contex
         return new self(new CronExpression(self::parseHashed($expression, $context)));
     }
 
+    public function jitter(int $value = 60): self
```
I kept this as a _builder_ method but can add as a constructor argument if desired.

javiereguiluz replied:
Just wondering if we could rename the method argument from `$value` to `$seconds` or `$numOfSeconds` or something that makes it self-explanatory. Thanks.

kbond replied:
Good idea. I've renamed to `$maxSeconds`.

fabpot commented:
I'm wondering if we should do this at a higher level. Having a jitter is probably a common need for most triggers.

kbond commented:
Yes, I was thinking that also.

kbond commented:
> I'm wondering if we should do this at a higher level. Having a jitter is probably a common need for most triggers.

I've updated this PR to add a `JitterTrigger` that can add a _jitter_ to any trigger. Also, added `RecurringMessage::withJitter()` to help with DX.

fabpot commented:
Thank you @kbond.
